### DISCUSSION TITLE
what's the different between "none" and "passthrough" log driver?

### DISCUSSION THREAD

**Discussion by GongT** [containers/podman#23928] [Q&A] [answered]
I found a new "passthrough" in man page, I used "none" for a long time.

I can't understand what is it. An alias?

**Comment by gabrielfrimodig** [comment]
`"none"` and `"passthrough"` are different log drivers. `none` completely disables logging so no logs are captured or stored, while `passthrough` sends logs directly to stdout/stderr withoug storing them. You see them in real-time but they are not stored.

So, use `none` if you want no logs at all and `passthrough` if you want to see live logs without keeping them.

**Reply by GongT** [reply]
But I use `none` several years, `none` log print program's output to my terminal, without storing them (`podman logs` show an error).

**Comment by Luap99** [comment, marked as answer]
if you run `podman run -d` you will see the difference more clearly, passthrough is actually passing down the actual stdio fds down in the container.
None on the other only doesn't log the output to somehwere so it cannot be replayed later, you can still use commands like podman attach to attach later and/or multiple times without issues all all the stdio is multiplexed in conmon. That is just not possible with passthrough as the fds only exists once.

**Reply by GongT** [reply]
Oh!!! really simple and clear, thank you!
